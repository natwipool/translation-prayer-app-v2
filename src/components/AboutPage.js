import React from 'react';

const AboutPage = () => (
  <div>
    <div className="page-header">
      <div className="content-container-body">
        <h2 className="page-header__title">เกี่ยวกับ</h2>
      </div>
    </div>
    <div className="content-container-body">
      <img className="cover-image" src="/images/about.jpg" />
      <div className="text-container">
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent
          tempus nisl odio, in sollicitudin nunc lacinia et. Sed molestie, nibh
          consectetur pulvinar pulvinar, urna ligula mattis tellus, quis
          molestie orci lectus eget libero. Curabitur ac ex turpis. Curabitur
          ante nunc, semper sed imperdiet quis, vehicula eget eros. Fusce id
          fermentum dolor. Ut gravida mauris enim, non placerat eros blandit
          nec. Nulla malesuada urna ut felis maximus, at luctus lectus
          hendrerit. Interdum et malesuada fames ac ante ipsum primis in
          faucibus. Pellentesque ultricies quam quis elit ultricies, ut ultrices
          sapien fermentum.
        </p>
        <p>
          Icons made by{' '}
          <a
            href="https://www.flaticon.com/authors/zlatko-najdenovski"
            title="Zlatko Najdenovski"
          >
            Zlatko Najdenovski
          </a>{' '}
          from{' '}
          <a href="https://www.flaticon.com/" title="Flaticon">
            www.flaticon.com
          </a>{' '}
          is licensed by{' '}
          <a
            href="http://creativecommons.org/licenses/by/3.0/"
            title="Creative Commons BY 3.0"
            target="_blank"
          >
            CC 3.0 BY
          </a>
        </p>
      </div>
    </div>
  </div>
);

export default AboutPage;
