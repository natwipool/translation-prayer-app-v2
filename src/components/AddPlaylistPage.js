import React from 'react';
import { connect } from 'react-redux';
import PlaylistForm from './PlaylistForm';
import { addPlaylist } from '../actions/playlists';

export class AddPlaylistPage extends React.Component {
  onSubmit = playlist => {
    this.props.addPlaylist(playlist);
    this.props.history.push('/playlists');
  };
  render() {
    return (
      <div>
        <h3>สร้างรายการสวดมนต์</h3>
        <PlaylistForm onSubmit={this.onSubmit} />
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  addPlaylist: playlist => dispatch(addPlaylist(playlist))
});

export default connect(undefined, mapDispatchToProps)(AddPlaylistPage);
