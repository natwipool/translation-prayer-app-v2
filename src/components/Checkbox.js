import React from 'react';

const Checkbox = props => {
  const onCheckboxChange = e => {
    const list = e.target.value;

    if (e.target.checked) {
      props.onAddLists(list);
    } else if (!e.target.checked) {
      props.onRemoveLists(list);
    }
  };

  return (
    <div>
      <label htmlFor={props.precept} className="checkbox">
        <input
          type="checkbox"
          id={props.precept}
          checked={props.lists.find(list => list === props.precept)}
          value={props.precept}
          onChange={onCheckboxChange}
        />
        <div className="checkbox-content">
        <span />
          
            <p className="checkbox-content__title">{props.precept}</p>
            
            {
              props.lists.indexOf(props.precept) > -1 &&
              <p className="checkbox-content__number">{props.lists.indexOf(props.precept) + 1}</p>
            } 
          
        </div>
      </label>
    </div>
  );
};

export default Checkbox;
