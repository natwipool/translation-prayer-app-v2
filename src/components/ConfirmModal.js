import React from 'react';
import { connect } from 'react-redux';
import Modal from 'react-modal';
import { startRemovePlaylist } from '../actions/playlists';

export class ConfirmModal extends React.Component {
  onRemove = () => {
    this.props.startRemovePlaylist({ id: this.props.playlist.id });
    this.props.history.push('/playlists');
  };
  render() {
    return (
      <Modal
        isOpen={!!this.props.showConfirmModal}
        contentLabel="confirm-delete"
        onRequestClose={this.props.handleCloseModal}
        ariaHideApp={false}
        closeTimeoutMS={200}
        className="Modal"
      >
        <div className="confirm-modal-content">
          <h3 className="modal-confirm__title">ลบรายการ ?</h3>
          <h3 className="modal-confirm__title">
            {this.props.playlist.description}
          </h3>
        <div className="confirm-modal-group-button">
          <button
            className="button--secondary del-button"
            onClick={this.onRemove}
          >
            ตกลง
          </button>
          <button
            className="button--secondary edit-button"
            onClick={this.props.handleCloseModal}
          >
            ยกเลิก
          </button>
          </div>
        </div>
      </Modal>
    );
  }
}

const mapDispatchToProps = (dispatch, props) => ({
  startRemovePlaylist: id => dispatch(startRemovePlaylist(id))
});

export default connect(undefined, mapDispatchToProps)(ConfirmModal);
