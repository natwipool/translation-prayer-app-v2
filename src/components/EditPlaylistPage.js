import React from 'react';
import { connect } from 'react-redux';
import PlaylistForm from './PlaylistForm';
import { startEditPlaylist } from '../actions/playlists';

export class EditPlaylistPage extends React.Component {
  onSubmit = playlist => {
    this.props.startEditPlaylist(this.props.playlist.id, playlist);
    this.props.history.push(`/playlist/${this.props.playlist.id}`);
  };
  render() {
    return (
      <div>
        <h3>แก้ไขรายการสวดมนต์</h3>
        <PlaylistForm playlist={this.props.playlist} onSubmit={this.onSubmit} />
      </div>
    );
  }
}

const mapStateToProps = (state, props) => ({
  playlist: state.playlists.find(
    playlist => playlist.id === props.match.params.id
  )
});

const mapDispatchToProps = dispatch => ({
  startEditPlaylist: (id, playlist) => dispatch(startEditPlaylist(id, playlist))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditPlaylistPage);
