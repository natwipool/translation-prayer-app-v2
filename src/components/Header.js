import React from 'react';
import { connect } from 'react-redux';
import { Link, NavLink } from 'react-router-dom';
import { Navbar, Nav, NavItem } from 'react-bootstrap';
import { LinkContainer } from 'react-router-bootstrap';
import LoginPage from '../components/LoginPage';
import { startLogout } from '../actions/auth';
import { handleShowLoginModal } from '../actions/modal';

export const Header = props => (
  <header>
    <Navbar inverse collapseOnSelect>
      <div className="content-container">
        <Navbar.Header>
          <Navbar.Brand>
            <Link to="/">สวดมนต์แปล</Link>
          </Navbar.Brand>
          <Navbar.Toggle />
        </Navbar.Header>
        <Navbar.Collapse>
          <Nav>
            <LinkContainer to="/trans-prayers">
              <NavItem>สารบัญ</NavItem>
            </LinkContainer>
            <LinkContainer to="/playlists">
              <NavItem>บทสวดมนต์ของฉัน</NavItem>
            </LinkContainer>
            <LinkContainer to="/about">
              <NavItem>เกี่ยวกับ</NavItem>
            </LinkContainer>
          </Nav>
          <Nav pullRight>
            <NavItem>
              {!props.isAuthenticated ? (
                <button 
                  className="button header-button"
                  onClick={() => {
                    props.handleShowLoginModal(true);
                  }}
                >
                  เข้าสู่ระบบ
                </button>
              ) : (
                <button 
                  className="button header-button logout-button"
                  onClick={props.startLogout}
                >
                  ออกจากระบบ
                </button>
              )}
            </NavItem>
          </Nav>
        </Navbar.Collapse>
      </div>
    </Navbar>
    <LoginPage
      showLoginModal={!!props.loginModal}
      handleHideLoginModal={() => {
        props.handleShowLoginModal();
      }}
    />
  </header>
);

const mapStateToProps = state => ({
  isAuthenticated: !!state.auth.uid,
  loginModal: state.modal.showLoginModal
});

const mapDispatchToProps = dispatch => ({
  startLogout: () => dispatch(startLogout()),
  handleShowLoginModal: isShow => dispatch(handleShowLoginModal(isShow))
});

export default connect(mapStateToProps, mapDispatchToProps)(Header);
