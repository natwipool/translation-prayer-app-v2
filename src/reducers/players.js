const defaultStatePlayerReducer = {
  isPlaying: false,
  index: 0
};

export default (state = defaultStatePlayerReducer, action) => {
  switch (action.type) {
    case 'IS_PLAYING_TOGGLE':
      return {
        ...state,
        isPlaying: !state.isPlaying,
      };
    case 'SET_PLAYING':
      return {
        ...state,
        isPlaying: true
      }
    case 'INCREMENT_INDEX':
      return {
        ...state,
        index: state.index + 1
      };
    case 'DECREMENT_INDEX':
      return {
        ...state,
        index: state.index - 1
      };
    case 'SET_INDEX':
      return {
        ...state,
        index: action.index
      }
    default:
      return state;
  }
};