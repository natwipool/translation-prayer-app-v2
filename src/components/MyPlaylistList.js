import React from 'react';
import { connect } from 'react-redux';
import { setIndex, setPlaying, closePlayer } from '../actions/players';
import PlayerOptions from './PlayerOptions';
import MyPlaylistListItem from './MyPlaylistListItem';
import Player from './Player';
import LyricsPage from './LyricsPage';
import getPlaylistData from '../utils/get-playlist-data';

export class MyPlaylistList extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      openModal: false
    };
  }

  handleOpenModal = () => {
    this.setState(() => ({ openModal: true }));
    this.props.setIndex(this.props.players.index);
  };

  handleCloseModal = () => {
    this.setState(() => ({ openModal: false }));
  };

  onPlayByIndex = index => {
    this.props.setIndex(index);
    this.props.setPlaying(true);
    this.setState(() => ({ openModal: true }));
  };

  onClosePlayer = () => {
    this.props.closePlayer();
  };

  render() {
    return (
      <div
        className={
          'content-container-body ' +
          (!this.props.players.closePlayer && 'content-container-player')
        }
      >
        <LyricsPage
          playlists={this.props.playlists}
          index={this.props.players.index}
          openModal={this.state.openModal}
          handleCloseModal={this.handleCloseModal}
        />

        <div className="list-header">
          <PlayerOptions onClosePlayer={this.onClosePlayer} />
        </div>
        {this.props.playlists.map((playlist, index) => (
          <div key={index} className="list-item">
            <MyPlaylistListItem {...playlist} />
            <button
              className="music-button"
              onClick={() => {
                this.onPlayByIndex(index);
              }}
            >
              {this.props.players.closePlayer ? (
                this.state.openModal && this.props.players.index === index ? (
                  <img src="/images/open-book.png" />
                ) : (
                  <img src="/images/book.png" />
                )
              ) : this.props.players.isPlaying &&
              this.props.players.index === index ? (
                this.state.openModal ? (
                  <img src="/images/open-book.png" />
                ) : (
                  <img src="/images/book.png" />
                )
              ) : (
                <img src="/images/play.png" />
              )}
            </button>
          </div>
        ))}
        {!this.props.players.closePlayer && (
          <Player playlists={this.props.playlists} />
        )}
      </div>
    );
  }
}

const mapStateToProps = (state, props) => ({
  playlists: getPlaylistData(state.transPrayers, props.lists),
  players: state.players
});

const mapDispatchToProps = dispatch => ({
  setIndex: index => dispatch(setIndex(index)),
  setPlaying: boolean => dispatch(setPlaying(boolean)),
  closePlayer: () => dispatch(closePlayer())
});

export default connect(mapStateToProps, mapDispatchToProps)(MyPlaylistList);
