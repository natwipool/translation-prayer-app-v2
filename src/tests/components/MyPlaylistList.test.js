import React from 'react';
import { shallow } from 'enzyme';
import { MyPlaylistList } from '../../components/MyPlaylistList';
import transPrayers from '../fixtures/transPrayers';
import players from '../fixtures/players';

let setIndex, setPlaying, wrapper;

beforeEach(() => {
  setIndex = jest.fn();
  setPlaying = jest.fn();

  wrapper = shallow(<MyPlaylistList
    setIndex={setIndex}
    setPlaying={setPlaying}
    playlists={transPrayers}
    players={players.notPlaying}
  />);
});

test('should render MyPlaylistList with playlists', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should handle onPlayByIndex correctly', () => {
  wrapper.find('button').at(0).simulate('click');
  expect(setIndex).toHaveBeenLastCalledWith(0);
  expect(setPlaying).toHaveBeenLastCalledWith(true);
});