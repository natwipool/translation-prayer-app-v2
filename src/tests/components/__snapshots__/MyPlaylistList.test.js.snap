// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MyPlaylistList correctly 1`] = `
<div
  className="content-container-body content-container-player"
>
  <Connect(LyricsPage)
    handleCloseModal={[Function]}
    index={0}
    openModal={false}
    playlists={
      Array [
        Object {
          "category": "บทสวดมนต์ภาคเย็น",
          "duration": 155.99425,
          "filename": "11_เขมาเขมสรณคมนปริทีปิกาคาถา.mp3",
          "id": "11",
          "lyrics": Array [
            "abc",
            "dfg",
          ],
          "precept": "เขมาเขมสรณคมนปริทีปิกาคาถา",
        },
        Object {
          "category": "บทสวดมนต์ภาคเย็น",
          "duration": 99.99425,
          "filename": "12_อริยธนคาถา.mp3",
          "id": "12",
          "lyrics": Array [
            "123",
            "456",
          ],
          "precept": "อริยธนคาถา",
        },
        Object {
          "category": "บทสวดมนต์ภาคเย็น",
          "duration": 212.99425,
          "filename": "13_ติลักขณาทิคาถา.mp3",
          "id": "13",
          "lyrics": Array [
            "789",
            "123",
          ],
          "precept": "ติลักขณาทิคาถา",
        },
      ]
    }
  />
  <div
    className="list-header"
  >
    <PlayerOptions
      onClosePlayer={[Function]}
    />
  </div>
  <div
    className="list-item"
    key="0"
  >
    <MyPlaylistListItem
      category="บทสวดมนต์ภาคเย็น"
      duration={155.99425}
      filename="11_เขมาเขมสรณคมนปริทีปิกาคาถา.mp3"
      id="11"
      lyrics={
        Array [
          "abc",
          "dfg",
        ]
      }
      precept="เขมาเขมสรณคมนปริทีปิกาคาถา"
    />
    <button
      className="music-button"
      onClick={[Function]}
    >
      <img
        src="/images/play.png"
      />
    </button>
  </div>
  <div
    className="list-item"
    key="1"
  >
    <MyPlaylistListItem
      category="บทสวดมนต์ภาคเย็น"
      duration={99.99425}
      filename="12_อริยธนคาถา.mp3"
      id="12"
      lyrics={
        Array [
          "123",
          "456",
        ]
      }
      precept="อริยธนคาถา"
    />
    <button
      className="music-button"
      onClick={[Function]}
    >
      <img
        src="/images/play.png"
      />
    </button>
  </div>
  <div
    className="list-item"
    key="2"
  >
    <MyPlaylistListItem
      category="บทสวดมนต์ภาคเย็น"
      duration={212.99425}
      filename="13_ติลักขณาทิคาถา.mp3"
      id="13"
      lyrics={
        Array [
          "789",
          "123",
        ]
      }
      precept="ติลักขณาทิคาถา"
    />
    <button
      className="music-button"
      onClick={[Function]}
    >
      <img
        src="/images/play.png"
      />
    </button>
  </div>
  <Connect(Player)
    playlists={
      Array [
        Object {
          "category": "บทสวดมนต์ภาคเย็น",
          "duration": 155.99425,
          "filename": "11_เขมาเขมสรณคมนปริทีปิกาคาถา.mp3",
          "id": "11",
          "lyrics": Array [
            "abc",
            "dfg",
          ],
          "precept": "เขมาเขมสรณคมนปริทีปิกาคาถา",
        },
        Object {
          "category": "บทสวดมนต์ภาคเย็น",
          "duration": 99.99425,
          "filename": "12_อริยธนคาถา.mp3",
          "id": "12",
          "lyrics": Array [
            "123",
            "456",
          ],
          "precept": "อริยธนคาถา",
        },
        Object {
          "category": "บทสวดมนต์ภาคเย็น",
          "duration": 212.99425,
          "filename": "13_ติลักขณาทิคาถา.mp3",
          "id": "13",
          "lyrics": Array [
            "789",
            "123",
          ],
          "precept": "ติลักขณาทิคาถา",
        },
      ]
    }
  />
</div>
`;
